### PR TITLE
Network/rnd icm refactoring

### PR DESCRIPTION
**Pull request by leonard-q**
:star2: Hello! Thanks for contributing JORLDY! 

### Checklist 

Please check if you consider the following items. 

- [V] My code follows the style guidelines of this project [contributing](https://github.com/kakaoenterprise/JORLDY/blob/master/CONTRIBUTING.md)
- [V] My code follows the [naming convention](https://github.com/kakaoenterprise/JORLDY/blob/master/docs/Naming_convention.md) of documentation
- [V] I have commented my code, particularly in hard-to-understand areas
- [V] My changes generate no new warnings or errors 

### Types of changes 

**Please describe the types of changes! (ex. Bugfix, New feature, Documentation, ...)**
Refactoring

### Test Configuration

- OS: Linux Ubuntu
- Python version: 3.9
- Additional libraries: 

### Description

**Please describe the details of your contribution**
RND and ICM networks are refactorized.

### CODE REVIEW

File: jorldy/core/network/icm.py [hunk 1]
```diff
@@ -4,6 +4,169 @@
 from .rnd import *
```
Comment (ramanuzan) [inline jorldy/core/network/icm.py:4]: rnd.py 의 함수들과 겹치는 이름들이 많아서, * 말고 직접 클래스, 함수 이름을 적어주는게 좋을 것 같습니다.

File: jorldy/core/network/icm.py [hunk 2]
```diff
@@ -4,6 +4,169 @@
...
 
 
+def mlp_head_weight(D_in, D_hidden, feature_size):
+    fc1 = torch.nn.Linear(D_in, D_hidden)
+    fc2 = torch.nn.Linear(D_hidden, feature_size)
+
+    return fc1, fc2
```
Comment (ramanuzan) [inline jorldy/core/network/icm.py:11]: 이런 방법으로 함수화(mlp_batch_norm, cnn_head_weight, cnn_batch_norm, ..)를 시켜서 재사용을 하다보니, 선언 변수들이 많아지면 코드라인 증가가 불가피한 것 같습니다.

레이어 클래스들 같은 경우에는 함수 인자로 인스턴스를 받아서 안에서 선언해주는건 어떨까요?

예시)
```
def define_var(instance, val):
    instance.var = val

class A:
    def __init__(self):
        define_var(self, 3)

a = A()
print(a.var)
========
3
```

File: jorldy/core/network/icm.py [hunk 3]
```diff
@@ -34,30 +197,15 @@ def __init__(
...
+        self.fc1, self.fc2 = mlp_head_weight(D_in, D_hidden, feature_size)
+        self.forward_fc1, self.forward_fc2, self.forward_loss = forward_weight(
+            feature_size, D_hidden, D_out, action_type
+        )
+        self.inverse_fc1, self.inverse_fc2, self.inverse_loss = inverse_weight(
+            feature_size, D_hidden, D_out, action_type
+        )
 
-            self.bn1_next = torch.nn.BatchNorm1d(D_hidden)
+        self.bn1, self.bn2, self.bn1_next = mlp_batch_norm(D_hidden, feature_size)
```
Comment (ramanuzan) [inline jorldy/core/network/icm.py:208]: self.batch_norm과 상관없이 해당 레이어들이 생성되는거 같아요.

아마 mlp_head 함수 때문인거 같은데요. mlp_head의 bn 들을 None으로 default로 잡아주고,
mlp_head를 self.batch_norm에 따라 다르게 실행하면 좋을 것 같습니다.

File: jorldy/core/network/icm.py (hunk 1 above)
Comment (leonard-q) [reply jorldy/core/network/icm.py:4]: 아마 normalize_obs 함수 하나만 불러오는 듯 하여 해당 함수 추가했습니다
